plugins/hypervisors/kvm/src/com/cloud/hypervisor/kvm/resource/LibvirtDomainXMLParser.java

```diff
@@ -237,6 +241,23 @@ public boolean parseDomainXML(String domXML) {
                 rngDefs.add(def);
             }
 
+            NodeList watchDogs = devices.getElementsByTagName("watchdog");
+            for (int i = 0; i < watchDogs.getLength(); i++) {
+                WatchDogDef def = null;
+                Element watchDog = (Element)watchDogs.item(i);
+                String action = watchDog.getAttribute("action");
+                String model = watchDog.getAttribute("model");
+
+                if (Strings.isNullOrEmpty(action)) {
+                    def = new WatchDogDef(WatchDogModel.valueOf(model.toUpperCase()));
+                } else {
+                    def = new WatchDogDef(WatchDogAction.valueOf(action.toUpperCase()),
+                                          WatchDogModel.valueOf(model.toUpperCase()));
+                }
+
+                watchDogDefs.add(def);
+            }
```

Comment:
Shouldn't it have a test if the 'model' is null/empty/blank (as the one applied to the 'action')? Thanks @wido!

Reply:
I think by default the action is None, so it might work, however, let's confirm with @wido ?

Reply:
I can do a check for that, but it's libvirt which will always return a model there. I'll add a if-statement around it, but that shouldn't make a huge difference. Thanks for the feedback!

Reply:
Assuming NPE won't happen on line 244, we can accept this per @wido 's comment (I've not personally verified this)